EmbeddedScripts.CSharp/CompiledCodeRunner.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using EmbeddedScripts.Shared;
+
+namespace EmbeddedScripts.CSharp
+{
+    public class CompiledCodeRunner : ICodeRunner
+    {
+        public CompiledCodeRunner(string code, Container container)
+        {
+            var usingLines = ResolvingCodeGenerator.GenerateUsingLines(container);
+            var resolveLines = ResolvingCodeGenerator.GenerateResolveLines(container);
+
+            var codeForCompilation = $@"
+using {typeof(Container).Namespace};
+{usingLines}
+
+public class {ClassName}
+{{
+    public void {MethodName}({nameof(Container)} Container) 
+    {{ 
+       {resolveLines}
+       {code}
+    }} 
+}}";
+
+            Container = container;
+            Compilation = new CodeCompiler(codeForCompilation, Container).Compilation;
+        }
+
+        public async Task RunAsync()
+        {
+            var assembly = LoadAssemblyFomCompilation(Compilation);
+            var type = assembly.GetType(ClassName);
+            var instance = Activator.CreateInstance(type);
+
+            await Task.Run(() =>
+                type.InvokeMember(
+                    MethodName,
+                    BindingFlags.Default | BindingFlags.InvokeMethod,
+                    null,
+                    instance,
+                    new object[] { Container }
```

Comment:
A lot of low level code. Maybe add more abstractions? Do we need them?

Reply:
Not sure if we needed it here - it's internal reflection-based implementation after all